aiida_mlip/calculations/singlepoint.py

```diff
@@ -0,0 +1,273 @@
+        )
+        spec.exit_code(
+            307,
+            "ERROR_CONTENT_OUTPUT_FILES",
+            message="The output file does not contain the right content",
+        )
+
+    @classmethod
+    def validate_inputs(
+        cls, inputs: dict, port_namespace: aiida.engine.processes.ports.PortNamespace
+    ) -> Union[str, None]:
+        """
+        Check if the inputs are valid.
+
+        Parameters
+        ----------
+        inputs : dict
+            The inputs dictionary.
+
+        port_namespace : `aiida.engine.processes.ports.PortNamespace`
+            An instance of aiida's `PortNameSpace`.
+
+        Returns
+        -------
+        str or None
+            Error message if validation fails, None otherwise.
+        """
+        # Wrapping processes may choose to exclude certain input ports
+        # If the ports have been excluded, skip the validation.
+        if any(key not in port_namespace for key in ("calctype", "structure")):
+            return None
+
+        for key in ("calctype", "structure"):
+            if key not in inputs:
+                return f"required value was not provided for the `{key}` namespace."
+
+        valid_calctypes = {"singlepoint", "geom opt"}
+        if "calctype" in inputs:
+            if str(inputs["calctype"].value) not in valid_calctypes:
+                return f"The 'calctype' must be one of {valid_calctypes}, \
+                    but got '{inputs['calctype']}'."
+
+        if "input_filename" in inputs:
+            if not str(inputs["input_filename"].value).endswith(".cif"):
+                return "The parameter 'input_filename' must end with '.cif'"
+
+        # If both structure and calctype are provided, return None
+        return None
+
+    # pylint: disable=too-many-locals
+    def prepare_for_submission(
+        self, folder: aiida.common.folders.Folder
+    ) -> datastructures.CalcInfo:
+        """
+        Create the input files for the `Calcjob`.
+
+        Parameters
+        ----------
+        folder : aiida.common.folders.Folder
+            An `aiida.common.folders.Folder` to temporarily write files on disk.
+
+        Returns
+        -------
+        aiida.common.datastructures.CalcInfo
+            An instance of `aiida.common.datastructures.CalcInfo`.
+        """
+        # Create needed inputs
+        # Define architecture from model if model is given,
+        # otherwise get architecture from inputs and download default model
+        architecture = (
+            str((self.inputs.model).architecture)
+            if self.inputs.model
+            else str(self.inputs.architecture.value)
+        )
+        model_path = (
+            str((self.inputs.model).filepath)
+            if self.inputs.model
+            else ModelData.download(
+                "https://github.com/stfc/janus-core/raw/main/tests/models/mace_mp_small.model",  # pylint:disable=line-too-long
+                architecture,
+            ).filepath
+        )
```

Comment:
```suggestion
        if self.inputs.model:
            model_path = self.inputs.model.filepath
        else:
            model_path = ModelData.download(
                "https://github.com/stfc/janus-core/raw/main/tests/models/mace_mp_small.model",  # pylint:disable=line-too-long
                architecture,
            ).filepath
```
Somewhat down to personal preference, but I generally avoid "one-liner" if/else if it spreads over multiple lines.

**Note**: I've also removed `str` conversions, as I don't think they're needed (tests pass for me) but that may be wrong